### ISSUE TITLE
[Feature Request] Zoom-in without reloading data

### ISSUE THREAD

**Issue by iShift** [grafana/grafana#10144]
It would bee cool if you can zoom-in without data-reloading (do that on client side) possible implementations - add dashboard setting and when you get stats first for 1 month (1.11.2017 - 12.11.2017) you get all dataset (as normal query) that when you change range to 1.11.2017-2.11.2017 you get data without new request to datasource - just filtering result on client side. 

That feature can dramatic speed up heavy calculation

**Comment by bergquist** [comment]
Just curious. What data source are you using?

Changing the time range (zooming) changes some the interval variables used to decide how many data points that can fit into a panel. So in the monthly query Grafana might ask for data every 1h while in the daily query Grafana might ask for data every 10min. 

Right now I don't see us supporting this behavior but I'll leave this issue open for a few days in case someone else has other opinions on the matter.

**Comment by iShift** [comment]
@bergquist i use ClickHouse datasource, i have more than 1PB data for group by, if i change time i make another group by with 1PB data =)

**Comment by bergquist** [comment]
ic :) 

No caching support?

I still think that's an edge case and probably, not something we want to support right now. But I'll leave this issue open if more people have the same problem.

**Comment by iShift** [comment]
How i can cache different sql query?)

**Comment by daniellee** [comment]
@iShift have you seen: https://github.com/Vertamedia/chproxy It has some caching support.

**Comment by iShift** [comment]
@daniellee Yep, i use it, but when you want to see only one day from month graph - it is new query, if we can have zoom in in grafana (based on data that we already have) - it would work blazing fast.

**Comment by gelicia** [comment]
Reopening as we had some more interest in this on the community slack.

**Comment by aangelisc** [comment]
I'm also going to add dashboards to this issue to gather more thoughts on this matter. I don't think this feature request should be implemented at the individual data source level, and I believe there is a fair amount of complexity surrounding how it may function.

Something that was mentioned in the community Slack, which is important to consider, is how the interval variable may affect the number of data points displayed. Zooming in on a time range but only filtering the data points may lead to missing data (or an incomplete picture).

Community Slack thread [here](https://grafana.slack.com/archives/C0Y4TLW74/p1748933946238219) for reference.
